### PR TITLE
Clean up HashAgg

### PR DESCRIPTION
**Pull request by hardikar**
Lots of minor changes. I'm cleaning things up as I implement a growing hash table.

1. Fix indentation & spelling mistakes wherever I saw them.
2. Remove some unused variables & dead code
3. Some calculations (which will be used in multiple places) are converted to macros : BLOOMVAL, BUCKET_IDX.
4. Chain statistics were shown only if we spilled - but it would be nice to know this even when the table was in memory.
5. Fix a small performance issue (because we didn't reset bloom filters while spilling).
6. Moved HashAggState to AggState from the hash table since it is referenced in nodeAgg.c and renamed that struct since it was very ambiguous.

For the 4th point, see the additional line in EXPLAIN ANALYZE that shows the chain length and number of buckets : 
```
shardikar=# explain analyze select i, count(*) from bar group by i;
                                                 QUERY PLAN
------------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)  (cost=1.01..1.03 rows=1 width=12)
   Rows out:  1 rows at destination with 11 ms to end, start offset by 9.732 ms.
   ->  HashAggregate  (cost=1.01..1.03 rows=1 width=12)
         Group By: i
         Rows out:  1 rows (seg0) with 6.699 ms to first row, 7.830 ms to end, start offset by 12 ms.
         Executor memory:  8233K bytes avg, 8281K bytes max (seg0).
         (seg0)   Hash chain length 1.0 avg, 1 max, using 1 of 524288 buckets.
         ->  Seq Scan on bar  (cost=0.00..1.01 rows=1 width=4)
               Rows out:  1 rows (seg0) with 0.276 ms to first row, 0.280 ms to end, start offset by 16 ms.
 Slice statistics:
   (slice0)    Executor memory: 322K bytes.
   (slice1)    Executor memory: 8318K bytes avg x 3 workers, 8366K bytes max (seg0).
 Statement statistics:
   Memory used: 128000K bytes
 Optimizer status: legacy query optimizer
 Total runtime: 21.601 ms
(16 rows)
```

### CODE REVIEW

File: src/backend/executor/execHHashagg.c [hunk 1]
```diff
@@ -86,6 +86,11 @@ typedef enum InputRecordType
 #define HAVE_FREESPACE(hashtable) \
    (GET_TOTAL_USED_SIZE(hashtable) < (hashtable)->max_mem)
 
+#define BLOOMVAL(hashkey) ((uint64)1) << (((hashkey) >> 23) & 0x3f);
+
+#define BUCKET_IDX(hashtable, hashkey) \
+		(((hashkey) >> (hashtable)->pshift) % (hashtable)->nbuckets)
+
```
Comment (hsyuan) [inline src/backend/executor/execHHashagg.c]: The L558 of the old file says that `/* Set nbuckets to the power of 2. */`, can we do bit manipulation here to get the index? say change `%` to `&`.

File: src/backend/executor/execHHashagg.c [hunk 2]
```diff
@@ -691,8 +697,10 @@ create_agg_hash_table(AggState *aggstate)
 	/* Initialize the hash buckets */
 	hashtable->nbuckets = hashtable->hats.nbuckets;
 	hashtable->total_buckets = hashtable->nbuckets;
-	hashtable->buckets = (HashAggEntry **)palloc0(hashtable->nbuckets * sizeof(HashAggEntry *));
-	hashtable->bloom = (uint64 *)palloc0(hashtable->nbuckets * sizeof(uint64));
+	hashtable->buckets = (HashAggEntry **) palloc0(hashtable->nbuckets * sizeof(HashAggEntry *));
+	hashtable->bloom = (uint64 *) palloc0(hashtable->nbuckets * sizeof(uint64));
+
+  hashtable->pshift = 0;
```
Comment (hsyuan) [inline src/backend/executor/execHHashagg.c]: indent

File: src/backend/executor/execHHashagg.c (hunk 1 above)
Comment (hardikar) [inline src/backend/executor/execHHashagg.c]: Yup - Already doing this in the growing hashtable implementation... so let's ignore this for now.

Comment (hsyuan) [inline src/backend/executor/execHHashagg.c]: OK, cool

File: src/backend/executor/execHHashagg.c [hunk 3]
```diff
@@ -376,7 +381,7 @@ static HashAggEntry *
 lookup_agg_hash_entry(AggState *aggstate,
 					  void *input_record,
 					  InputRecordType input_type, int32 input_size,
-					  uint32 hashkey, unsigned parent_hash_bit, bool *p_isnew)
+					  uint32 hashkey, bool *p_isnew)
 {
```
Comment (foyzur) [inline src/backend/executor/execHHashagg.c]: Please add parameter description in the function comment

File: src/backend/executor/execHHashagg.c [hunk 4]
```diff
@@ -1878,40 +1884,54 @@ agg_hash_next_pass(AggState *aggstate)
...
 		instr->workmemused = hashtable->mem_used;
+	}
+
+	return more;
+}
 
+/*
+ * Append hashtable statistics to explain buffer for EXPLAIN ANALYZE
+ */
+void
```
Comment (foyzur) [inline src/backend/executor/execHHashagg.c]: static please

File: src/backend/executor/execHHashagg.c [hunk 5]
```diff
@@ -1966,6 +1988,7 @@ void destroy_agg_hash_table(AggState *aggstate)
 	
 	if ( agg->aggstrategy == AGG_HASHED && aggstate->hhashtable != NULL )
 	{
+
```
Comment (foyzur) [inline src/backend/executor/execHHashagg.c]: please remove this space

**Pull request merged**